File: frameowkrs/typescript-cypress/pages/checkout_page.ts
export class CheckoutPage {
  placeOrder(): Cypress.Chainable<number> {
    cy.get('[data-testid="checkout-use-fake-name-email"]').check();
    cy.get('[data-testid="checkout-use-fake-address"]').check();
    cy.get('[data-testid="checkout-use-testing-card"]').check();
    cy.get('[data-testid="checkout-submit"]').click();
    return cy.url().should('match', /\/orders\/\d+$/).then((url) => {
      const orderId = Number(url.match(/\/orders\/(\d+)$/)?.[1]);
      if (!orderId) throw new Error('Order ID was not present in the order detail URL');
      return orderId;
    });
  }
}

File: frameowkrs/python-playwright/pages/checkout_page.py
import re

from playwright.sync_api import Page, expect


class CheckoutPage:
    def __init__(self, page: Page) -> None:
        self.page = page

    def place_order(self) -> int:
        self.page.get_by_test_id("checkout-use-fake-name-email").check()
        self.page.get_by_test_id("checkout-use-fake-address").check()
        self.page.get_by_test_id("checkout-use-testing-card").check()
        self.page.get_by_test_id("checkout-submit").click()
        expect(self.page).to_have_url(re.compile(r"/orders/\d+$"))
        match = re.search(r"/orders/(\d+)$", self.page.url)
        if not match:
            raise AssertionError("Order ID was not present in the order detail URL")
        return int(match.group(1))
